pkg/hostpath/mount.go

```diff
@@ -0,0 +1,143 @@
+package hostpath
+
+import (
+	"fmt"
+	"os"
+	"strings"
+
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+
+	"k8s.io/utils/mount"
+)
+
+type FileSystemMounter interface {
+	makeFSMounts(anchorDir, intermediateBindMountDir, kubeletTargetDir string, mounter mount.Interface) error
+	removeFSMounts(anchorDir, intermediateBindMountDir, kubeletTargetDir string, mount mount.Interface) error
+}
+
+// ReadWriteMany high level details:
+//
+// This is our original landing spot wrt mounting the file system this driver manipulates
+// to where the location the kubelet has allocated for the CSI volume in question.
+//
+// We go straight from our tmpfs anchor to kubelet's target directory.  No bind mounts.
+// But this approach only works if the K8s CSIVolumenSource set readOnly to false.  If readOnly
+// is set to true, the underlying mount mechanics between our call to m.mounter.Mount and what
+// the kubelet does for the Pod results in the use of xfs for the filesystem and an inability for the
+// Pod to read what we have mounted.
+//
+// Additional details:
+//
+// So so our intent here is to have a separate tmpfs per pod; through experimentation
+// and corroboration with OpenShift storage SMEs, a separate tmpfs per pod
+// - ensures the kubelet will handle SELinux for us. It will relabel the volume in "the right way" just for the pod
+// - otherwise, if pods share the same host dir, all sorts of warnings from the SMEs
+// - and the obvious isolation between pods that implies
+// We cannot do read-only on the mount since we have to copy the data after the mount, otherwise we get errors
+// that the filesystem is readonly.
+// However, we can restart this driver, leave up any live Pods with our volume, and then still update the content
+// after this driver comes backup.
+// The various bits that work in concert to achieve this
+// - the use of emptyDir with a medium of Memory in this drivers Deployment is all that is needed to get tmpfs
+// - do not use the "bind" option, that reuses existing dirs/filesystems vs. creating new tmpfs
+// - without bind, we have to specify an fstype of tmpfs and path for the mount source, or we get errors on the
+//   Mount about the fs not being  block access
+// - that said,  testing confirmed using fstype of tmpfs on hostpath/xfs volumes still results in the target
+//   being xfs and not tmpfs
+// - with the lack of a bind option, and each pod getting its own tmpfs we have to copy the data from our emptydir
+//   based location to the targetPath here ... that is handled in hostpath.go
+type ReadWriteMany struct {
+}
+
+func (m *ReadWriteMany) makeFSMounts(anchorDir, intermediateBindMountDir, kubeletTargetDir string, mounter mount.Interface) error {
+	options := []string{}
+	if err := mounter.Mount(anchorDir, kubeletTargetDir, "tmpfs", options); err != nil {
+		var errList strings.Builder
```

Comment:
nit: I don't think that strings.Builder is a List. Maybe a byte array?

Reply:
yeah this code is borrowed from the upstream k8s csi driver sample

strings.Builder.WriteString per its godoc appends content .... that is array / list enough for me ... I'm inclined to keep this as is